### PR TITLE
[Clang] update reasoned delete diagnostic kind to use Extension, making it pedantic only

### LINKED ISSUE #109311: Clang should not warn about =delete("reason") in C++17 mode while claiming to support it
Clang 19 seems to define `__cpp_deleted_function` even in pre-C++26 modes. However using it results in a warning:

```
void f() = delete
#if __cpp_deleted_function >= 202403L
    ("reason")
#endif
;
```

```
<source>:3:6: warning: '= delete' with a message is a C++2c extension [-Wc++26-extensions]
    3 |     ("reason")
      |      ^~~~~~~~
1 warning generated.
Compiler returned: 0
```

https://godbolt.org/z/Esqh5aPfb

It sounds a bit strange to claim support for a given language feature (by setting the corresponding feature-define macro), but then warn if that feature is used? Would it be possible, in pre-C++26 modes, to support the syntax and warn but _not_ set the macro?

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #109311

--- 

https://github.com/llvm/llvm-project/issues/109311#issuecomment-2422963686

### CODE REVIEW

**Comment by zwuis** [comment]
I think we should make these changes to all backported features.

File: clang/docs/ReleaseNotes.rst
Comment (AaronBallman) [inline clang/docs/ReleaseNotes.rst:467]:
```suggestion
- Clang now diagnoses ``= delete("reason")`` extension warnings only in pedantic mode rather than on by default. (#GH109311).
```

File: clang/test/Parser/cxx2c-delete-with-message.cpp
Comment (AaronBallman) [inline clang/test/Parser/cxx2c-delete-with-message.cpp:6]: It would be better to avoid the conditional logic and instead use `-verify` prefixes alone.

File: clang/test/Parser/cxx2c-delete-with-message.cpp
Comment (AaronBallman) [inline clang/test/Parser/cxx2c-delete-with-message.cpp:62]: When you make edits, please make sure the file ends with a newline again.

**Review by AaronBallman — COMMENTED** [review]
Thank you for working on this!

**Comment by AaronBallman** [comment]
> > I think we should make these changes to all backported features.
> 
> @zwuis @AaronBallman is it okay to handle that in a separate issue/PR?

Yeah, I think it's fine to handle separately.

**Review by AaronBallman — APPROVED** [review]
LGTM assuming precommit CI comes back green.

Btw, now would probably be a good time for you to get your own commit privileges, you've done a number of quality contributions already: https://llvm.org/docs/DeveloperPolicy.html#obtaining-commit-access

File: clang/docs/ReleaseNotes.rst
Comment (a-tarasyuk) [reply clang/docs/ReleaseNotes.rst]: @AaronBallman The CI checks have passed, if no further reviews are required, could you please merge it? thanks

Comment (AaronBallman) [reply clang/docs/ReleaseNotes.rst]: At this point, I'd prefer if you'd obtain commit privileges and land the changes yourself -- you have a number of patches under your belt, and it's better for folks to land their own changes in case of needing reverts, etc.

If there's a reason you don't want/can't get commit rights, feel free to let me know privately or otherwise, and I can certainly land the changes for you.

Comment (a-tarasyuk) [reply clang/docs/ReleaseNotes.rst]: @AaronBallman Got it, thanks.

**Pull request merged**
